src/workerd/api/node/url.c++

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+#include "url.h"
+
+#include <ada.h>
+
+namespace workerd::api::node {
+
+kj::ArrayPtr<const char> UrlUtil::domainToUnicode(kj::String domain) {
+  if (domain.size() == 0) {
+    return {};
+  }
+
+  auto out = ada::parse<ada::url_aggregator>("ws://x");
+  JSG_REQUIRE(out.has_value(), Error, "URL parsing failed"_kj);
+  if (!out->set_hostname({domain.begin(), domain.size()})) {
+    return {};
+  }
+  auto result = out->get_hostname();
+  return kj::ArrayPtr<const char>(result.data(), result.size());
+}
+
+kj::ArrayPtr<const char> UrlUtil::domainToASCII(kj::String domain) {
+  if (domain.size() == 0) {
+    return {};
+  }
+
+  auto out = ada::parse<ada::url_aggregator>("ws://x");
+  JSG_REQUIRE(out.has_value(), Error, "URL parsing failed"_kj);
+  if (!out->set_hostname({domain.begin(), domain.size()})) {
+    return {};
+  }
+
+  auto result = ada::idna::to_unicode(out->get_hostname());
```

Comment:
It's a bit odd that the toASCII function is invoking to_unicode. Were they swapped by accident? If not, please add a comment to explain what to_unicode does.